src/common/upgrade.rs

```diff
@@ -52,4 +52,23 @@ impl Upgrade {
     pub fn websocket() -> Upgrade {
         Upgrade(HeaderValue::from_static("websocket"))
     }
+
+    /// Constructs an `Upgrade: HTTP/2.0` header.
+    pub fn http2() -> Upgrade {
+        // must be uppercase, see
+        // <https://datatracker.ietf.org/doc/html/rfc7230#section-2.6>
+        Upgrade(HeaderValue::from_static("HTTP/2.0"))
```

Comment:
The [spec says](https://httpwg.org/specs/rfc7540.html#rfc.section.3.2) the header value would be `h2c`.

Reply:
Understood - I added an `h2c` constructor for that case and noted that ALPN/prior knowledge are overwhelmingly more common for the "http/2 over TLS" case.